### DISCUSSION TITLE
Typing can function on resource

### DISCUSSION THREAD

**Discussion by Wulfheart** [filamentphp/filament#5426] [Ideas]
```php
public static function can(string $action, ?Model $record = null): bool
```

I don't know what the plans are for v3 but I would generally wish for more typing support. E.g. the action here. I know that it can be `view`, `edit`, ... However, this has not good autocomplete support. Wouldn't it lead to more typesafety to add a backed enum? 
```php
public static function can(string|ActionEnum $action, ?Model $record = null): bool
```

**Comment by danharrin** [comment]
Unfortunately, unless we drop PHP 8 support in v3, we cant add enums into functions like this

**Reply by Wulfheart** [reply]
Maybe as a docblock parameter?

**Reply by danharrin** [reply]
We cant define the enums in the first place without PHP 8.1. There would be a language error
